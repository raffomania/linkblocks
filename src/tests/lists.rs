use sqlx::{Pool, Postgres};

use crate::tests::util::test_app::TestApp;

#[test_log::test(sqlx::test)]
async fn get_create_list(pool: Pool<Postgres>) -> anyhow::Result<()> {
    let mut app = TestApp::new(pool).await;
    app.create_test_user().await;
    app.login_test_user().await;

    let create_list = app.req().get("/lists/create").await.test_page().await;

    insta::assert_snapshot!(create_list.dom.htmls());

    Ok(())
}
