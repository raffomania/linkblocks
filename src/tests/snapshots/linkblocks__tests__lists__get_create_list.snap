---
source: src/tests/lists.rs
expression: create_list.dom.htmls()
---
  <head>
    <link rel="stylesheet" href="/assets/preflight.css" />
    <link rel="stylesheet" href="/assets/railwind.css" />
    <script src="/assets/htmx.1.9.9.js"></script>
    <meta name="color-scheme" content="dark" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>linkblocks</title>
  </head>
  <body class="w-full h-full text-gray-200 bg-neutral-800">
    <div class="flex-row-reverse h-full sm:flex">
      <main class="sm:overflow-y-auto sm:grow">
        <form action="/lists/create" method="POST" class="flex flex-col max-w-xl mx-4 mb-4 grow">
          <header class="mt-3 mb-4">
            <h1 class="text-xl font-bold">Create a list</h1>
          </header>
          <label for="title">Title</label>
          <input required="" name="title" type="text" value="" class="rounded py-1.5 px-3 mt-2 bg-neutral-900" />
          <label class="mt-4">Note<textarea name="content" placeholder="" value="" class="rounded py-1.5 px-3 mt-2 bg-neutral-900 block w-full"></textarea></label>
          <div class="mt-3 mb-5">
            <label><input type="checkbox" name="private" value="true" />Private</label>
          </div>
          <button type="submit" class="bg-neutral-300 py-1.5 px-3 text-neutral-900 rounded mt-4 self-end">
            Add List
          </button>
        </form>
      </main>
      <aside id="nav" class="bg-neutral-900 sm:max-w-[18rem] sm:w-1/3 sm:max-h-full flex flex-col sm:flex-col-reverse sm:border-r border-neutral-700 border-t sm:border-t-0">
        <div class="sm:overflow-y-auto sm:flex-1">
          <div class="sticky top-0 flex items-center justify-between px-2 pt-2 sm:top-0 bg-neutral-900">
            <h3 class="px-2 py-1 text-sm font-bold tracking-tight text-neutral-400">
              Lists
            </h3>
            <a href="/lists/create" class="block px-3 text-xl rounded hover:bg-neutral-800 text-neutral-400">+</a>
          </div>
          <ul class="pb-2">
            <li>
              <a class="block px-4 py-1 overflow-hidden text-ellipsis whitespace-nowrap hover:bg-neutral-800" href="/bookmarks/unsorted">Unsorted bookmarks</a>
            </li>
            <li>
              <a class="block px-4 py-1 overflow-hidden text-ellipsis whitespace-nowrap hover:bg-neutral-800 text-neutral-400" href="/lists/unpinned">Unpinned lists</a>
            </li>
          </ul>
        </div>
        <header class="sticky bottom-0 flex justify-between p-2 leading-8 bg-neutral-900">
          <a href="/" class="px-2 font-bold rounded hover:bg-neutral-800">testuser</a>
          <form action="/logout" method="post">
            <button class="rounded px-3 text-neutral-400 hover:bg-neutral-800">
              Logout
            </button>
          </form>
        </header>
      </aside>
    </div>
  </body>
